src/helpers/guards/isError.test.ts

```diff
@@ -0,0 +1,45 @@
+import { isError } from './isError';
+
+class TestError extends Error {
+	constructor() {
+		super('This is an error');
+		this.name = 'TestError';
+	}
+}
+
+class TestNonError {
+	readonly name: string;
+
+	constructor() {
+		this.name = 'TestNonError';
+	}
+}
+
+describe('isError', () => {
+	test('returns true for Error instances', () => {
+		expect(isError(new Error('This is an error'))).toBeTrue();
+	});
+
+	test('returns true for Error subclasses', () => {
+		expect(isError(new EvalError('This is an error'))).toBeTrue();
+		expect(isError(new RangeError('This is an error'))).toBeTrue();
+		expect(isError(new ReferenceError('This is an error'))).toBeTrue();
+		expect(isError(new SyntaxError('This is an error'))).toBeTrue();
+		expect(isError(new TestError())).toBeTrue();
+		expect(isError(new TypeError('This is an error'))).toBeTrue();
+	});
+
+	test.each`
```

Comment:
the h*cc is this syntax?

Reply:
This is a [Template Literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals). They can be used as function arguments like this to do really weird JavaScript magic. In this case, Jest uses this as their first-party solution for parameterized tests. I hate it, but it's well-documented Jest, and it works, lol